### PR TITLE
Continuously send events during e2e tests

### PR DESCRIPTION
**Pull request by pierDipi**
This PR adds configurations for a tool that sends events continuously 
(200 per second for 10 minutes) and detects events loss.

## Proposed Changes

- Continuously send events during e2e tests

### CODE REVIEW

**Comment by pierDipi** [comment]
```
Error from server (InternalError): error when creating "STDIN": Internal error occurred: failed calling webhook "webhook.eventing.knative.dev": Post https://eventing-webhook.knative-eventing.svc:443/defaulting?timeout=2s: EOF
```
issue: https://github.com/knative/pkg/issues/1509

/test pull-knative-sandbox-eventing-kafka-broker-integration-tests

**Comment by pierDipi** [comment]
First run: we have 3 events lost over 600, and the dispatcher failed to send those events because there are some _ConnectionRefused_ errors (without retries).

```json
--> "87af79c4-dda4-4670-919c-1f679076c46f" --> "87af79c4-dda4-4670-919c-1f679076c46f"
--> "8bd9d0de-70fd-4b12-893e-c2fb4b4b9ba4" --> "8bd9d0de-70fd-4b12-893e-c2fb4b4b9ba4"
--> "e180810b-9a5b-4473-9c68-56fba1251f72" --> "e180810b-9a5b-4473-9c68-56fba1251f72"

{
  "@timestamp": "2020-10-03T14:26:42.542Z",
  "@version": "1",
  "message": "subscriber sender failed to send record topic=knative-broker-knative-eventing-broker partition=4 offset=55",
  "logger_name": "dev.knative.eventing.kafka.broker.dispatcher.ConsumerRecordHandler",
  "thread_name": "vert.x-eventloop-thread-0",
  "level": "ERROR",
  "level_value": 40000,
  "stack_trace": "io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: sacura.knative-eventing.svc.cluster.local/10.27.245.104:80\nCaused by: java.net.ConnectException: Connection refused\n\tat java.base/sun.nio.ch.Net.pollConnect(Native Method)\n\tat java.base/sun.nio.ch.Net.pollConnectNow(Unknown Source)\n\tat java.base/sun.nio.ch.SocketChannelImpl.finishConnect(Unknown Source)\n\tat io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:330)\n\tat io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:334)\n\tat io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:702)\n\tat io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:650)\n\tat io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:576)\n\tat io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:493)\n\tat io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)\n\tat io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)\n\tat io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)\n\tat java.base/java.lang.Thread.run(Unknown Source)\n",
  "topic": "knative-broker-knative-eventing-broker",
  "partition": 4,
  "offset": 55,
  "event": {
    "data": "ImhlbGxvIg==",
    "id": "87af79c4-dda4-4670-919c-1f679076c46f",
    "source": "http://example.com/source",
    "type": "com.example.FullEvent",
    "subject": "topic",
    "time": {
      "offset": {
        "totalSeconds": 0,
        "id": "Z",
        "rules": {
          "fixedOffset": true,
          "transitions": [],
          "transitionRules": []
        }
      },
      "nano": 780000000,
      "year": 2020,
      "monthValue": 3,
      "dayOfMonth": 21,
      "hour": 12,
      "minute": 34,
      "second": 56,
      "month": "MARCH",
      "dayOfWeek": "SATURDAY",
      "dayOfYear": 81
    },
    "dataSchema": "http://example.com/schema",
    "dataContentType": "text/json",
    "specVersion": "V1",
    "extensionNames": [
      "exstring",
      "extime",
      "exint",
      "exurl",
      "exbinary",
      "exbool"
    ],
    "attributeNames": [
      "datacontenttype",
      "subject",
      "specversion",
      "id",
      "source",
      "time",
      "type",
      "dataschema"
    ]
  }
}
{
  "@timestamp": "2020-10-03T14:26:42.587Z",
  "@version": "1",
  "message": "dead letter queue sender failed to send record topic=knative-broker-knative-eventing-broker partition=4 offset=55",
  "logger_name": "dev.knative.eventing.kafka.broker.dispatcher.ConsumerRecordHandler",
  "thread_name": "vert.x-eventloop-thread-0",
  "level": "ERROR",
  "level_value": 40000,
  "stack_trace": "io.vertx.core.impl.NoStackTraceThrowable: no DLQ set\n",
  "topic": "knative-broker-knative-eventing-broker",
  "partition": 4,
  "offset": 55,
  "event": {
    "data": "ImhlbGxvIg==",
    "id": "87af79c4-dda4-4670-919c-1f679076c46f",
    "source": "http://example.com/source",
    "type": "com.example.FullEvent",
    "subject": "topic",
    "time": {
      "offset": {
        "totalSeconds": 0,
        "id": "Z",
        "rules": {
          "fixedOffset": true,
          "transitions": [],
          "transitionRules": []
        }
      },
      "nano": 780000000,
      "year": 2020,
      "monthValue": 3,
      "dayOfMonth": 21,
      "hour": 12,
      "minute": 34,
      "second": 56,
      "month": "MARCH",
      "dayOfWeek": "SATURDAY",
      "dayOfYear": 81
    },
    "dataSchema": "http://example.com/schema",
    "dataContentType": "text/json",
    "specVersion": "V1",
    "extensionNames": [
      "exstring",
      "extime",
      "exint",
      "exurl",
      "exbinary",
      "exbool"
    ],
    "attributeNames": [
      "datacontenttype",
      "subject",
      "specversion",
      "id",
      "source",
      "time",
      "type",
      "dataschema"
    ]
  }
}
{
  "@timestamp": "2020-10-03T14:26:43.628Z",
  "@version": "1",
  "message": "subscriber sender failed to send record topic=knative-broker-knative-eventing-broker partition=1 offset=61",
  "logger_name": "dev.knative.eventing.kafka.broker.dispatcher.ConsumerRecordHandler",
  "thread_name": "vert.x-eventloop-thread-0",
  "level": "ERROR",
  "level_value": 40000,
  "stack_trace": "io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: sacura.knative-eventing.svc.cluster.local/10.27.245.104:80\nCaused by: java.net.ConnectException: Connection refused\n\tat java.base/sun.nio.ch.Net.pollConnect(Native Method)\n\tat java.base/sun.nio.ch.Net.pollConnectNow(Unknown Source)\n\tat java.base/sun.nio.ch.SocketChannelImpl.finishConnect(Unknown Source)\n\tat io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:330)\n\tat io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:334)\n\tat io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:702)\n\tat io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:650)\n\tat io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:576)\n\tat io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:493)\n\tat io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)\n\tat io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)\n\tat io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)\n\tat java.base/java.lang.Thread.run(Unknown Source)\n",
  "topic": "knative-broker-knative-eventing-broker",
  "partition": 1,
  "offset": 61,
  "event": {
    "data": "ImhlbGxvIg==",
    "id": "e180810b-9a5b-4473-9c68-56fba1251f72",
    "source": "http://example.com/source",
    "type": "com.example.FullEvent",
    "subject": "topic",
    "time": {
      "offset": {
        "totalSeconds": 0,
        "id": "Z",
        "rules": {
          "fixedOffset": true,
          "transitions": [],
          "transitionRules": []
        }
      },
      "nano": 780000000,
      "year": 2020,
      "monthValue": 3,
      "dayOfMonth": 21,
      "hour": 12,
      "minute": 34,
      "second": 56,
      "month": "MARCH",
      "dayOfWeek": "SATURDAY",
      "dayOfYear": 81
    },
    "dataSchema": "http://example.com/schema",
    "dataContentType": "text/json",
    "specVersion": "V1",
    "extensionNames": [
      "exstring",
      "extime",
      "exint",
      "exurl",
      "exbinary",
      "exbool"
    ],
    "attributeNames": [
      "datacontenttype",
      "subject",
      "specversion",
      "id",
      "source",
      "time",
      "type",
      "dataschema"
    ]
  }
}
{
  "@timestamp": "2020-10-03T14:26:43.629Z",
  "@version": "1",
  "message": "dead letter queue sender failed to send record topic=knative-broker-knative-eventing-broker partition=1 offset=61",
  "logger_name": "dev.knative.eventing.kafka.broker.dispatcher.ConsumerRecordHandler",
  "thread_name": "vert.x-eventloop-thread-0",
  "level": "ERROR",
  "level_value": 40000,
  "stack_trace": "io.vertx.core.impl.NoStackTraceThrowable: no DLQ set\n",
  "topic": "knative-broker-knative-eventing-broker",
  "partition": 1,
  "offset": 61,
  "event": {
    "data": "ImhlbGxvIg==",
    "id": "e180810b-9a5b-4473-9c68-56fba1251f72",
    "source": "http://example.com/source",
    "type": "com.example.FullEvent",
    "subject": "topic",
    "time": {
      "offset": {
        "totalSeconds": 0,
        "id": "Z",
        "rules": {
          "fixedOffset": true,
          "transitions": [],
          "transitionRules": []
        }
      },
      "nano": 780000000,
      "year": 2020,
      "monthValue": 3,
      "dayOfMonth": 21,
      "hour": 12,
      "minute": 34,
      "second": 56,
      "month": "MARCH",
      "dayOfWeek": "SATURDAY",
      "dayOfYear": 81
    },
    "dataSchema": "http://example.com/schema",
    "dataContentType": "text/json",
    "specVersion": "V1",
    "extensionNames": [
      "exstring",
      "extime",
      "exint",
      "exurl",
      "exbinary",
      "exbool"
    ],
    "attributeNames": [
      "datacontenttype",
      "subject",
      "specversion",
      "id",
      "source",
      "time",
      "type",
      "dataschema"
    ]
  }
}
{
  "@timestamp": "2020-10-03T14:26:44.652Z",
  "@version": "1",
  "message": "subscriber sender failed to send record topic=knative-broker-knative-eventing-broker partition=2 offset=54",
  "logger_name": "dev.knative.eventing.kafka.broker.dispatcher.ConsumerRecordHandler",
  "thread_name": "vert.x-eventloop-thread-0",
  "level": "ERROR",
  "level_value": 40000,
  "stack_trace": "io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: sacura.knative-eventing.svc.cluster.local/10.27.245.104:80\nCaused by: java.net.ConnectException: Connection refused\n\tat java.base/sun.nio.ch.Net.pollConnect(Native Method)\n\tat java.base/sun.nio.ch.Net.pollConnectNow(Unknown Source)\n\tat java.base/sun.nio.ch.SocketChannelImpl.finishConnect(Unknown Source)\n\tat io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:330)\n\tat io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:334)\n\tat io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:702)\n\tat io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:650)\n\tat io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:576)\n\tat io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:493)\n\tat io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)\n\tat io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)\n\tat io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)\n\tat java.base/java.lang.Thread.run(Unknown Source)\n",
  "topic": "knative-broker-knative-eventing-broker",
  "partition": 2,
  "offset": 54,
  "event": {
    "data": "ImhlbGxvIg==",
    "id": "8bd9d0de-70fd-4b12-893e-c2fb4b4b9ba4",
    "source": "http://example.com/source",
    "type": "com.example.FullEvent",
    "subject": "topic",
    "time": {
      "offset": {
        "totalSeconds": 0,
        "id": "Z",
        "rules": {
          "fixedOffset": true,
          "transitions": [],
          "transitionRules": []
        }
      },
      "nano": 780000000,
      "year": 2020,
      "monthValue": 3,
      "dayOfMonth": 21,
      "hour": 12,
      "minute": 34,
      "second": 56,
      "month": "MARCH",
      "dayOfWeek": "SATURDAY",
      "dayOfYear": 81
    },
    "dataSchema": "http://example.com/schema",
    "dataContentType": "text/json",
    "specVersion": "V1",
    "extensionNames": [
      "exstring",
      "extime",
      "exint",
      "exurl",
      "exbinary",
      "exbool"
    ],
    "attributeNames": [
      "datacontenttype",
      "subject",
      "specversion",
      "id",
      "source",
      "time",
      "type",
      "dataschema"
    ]
  }
}
{
  "@timestamp": "2020-10-03T14:26:44.653Z",
  "@version": "1",
  "message": "dead letter queue sender failed to send record topic=knative-broker-knative-eventing-broker partition=2 offset=54",
  "logger_name": "dev.knative.eventing.kafka.broker.dispatcher.ConsumerRecordHandler",
  "thread_name": "vert.x-eventloop-thread-0",
  "level": "ERROR",
  "level_value": 40000,
  "stack_trace": "io.vertx.core.impl.NoStackTraceThrowable: no DLQ set\n",
  "topic": "knative-broker-knative-eventing-broker",
  "partition": 2,
  "offset": 54,
  "event": {
    "data": "ImhlbGxvIg==",
    "id": "8bd9d0de-70fd-4b12-893e-c2fb4b4b9ba4",
    "source": "http://example.com/source",
    "type": "com.example.FullEvent",
    "subject": "topic",
    "time": {
      "offset": {
        "totalSeconds": 0,
        "id": "Z",
        "rules": {
          "fixedOffset": true,
          "transitions": [],
          "transitionRules": []
        }
      },
      "nano": 780000000,
      "year": 2020,
      "monthValue": 3,
      "dayOfMonth": 21,
      "hour": 12,
      "minute": 34,
      "second": 56,
      "month": "MARCH",
      "dayOfWeek": "SATURDAY",
      "dayOfYear": 81
    },
    "dataSchema": "http://example.com/schema",
    "dataContentType": "text/json",
    "specVersion": "V1",
    "extensionNames": [
      "exstring",
      "extime",
      "exint",
      "exurl",
      "exbinary",
      "exbool"
    ],
    "attributeNames": [
      "datacontenttype",
      "subject",
      "specversion",
      "id",
      "source",
      "time",
      "type",
      "dataschema"
    ]
  }
}
```

**Comment by pierDipi** [comment]
Worked, retesting
/test pull-knative-sandbox-eventing-kafka-broker-integration-tests

**Comment by pierDipi** [comment]
2/2 worked, retesting
/test pull-knative-sandbox-eventing-kafka-broker-integration-tests

**Comment by pierDipi** [comment]
3/3 worked, retesting
/test pull-knative-sandbox-eventing-kafka-broker-integration-tests

**Comment by pierDipi** [comment]
worked, retesting
/test pull-knative-sandbox-eventing-kafka-broker-integration-tests

**Comment by pierDipi** [comment]
Maven error,
/retest

**Comment by pierDipi** [comment]
worked, retesting
/test pull-knative-sandbox-eventing-kafka-broker-integration-tests

**Comment by pierDipi** [comment]
worked, retesting
/test pull-knative-sandbox-eventing-kafka-broker-integration-tests

**Comment by pierDipi** [comment]
2/2
worked, retesting
/test pull-knative-sandbox-eventing-kafka-broker-integration-tests

**Comment by pierDipi** [comment]
3/3
[The job worked](https://00f74ba44bea4f9e18b59ff8beb78dd46ec7153624-apidata.googleusercontent.com/download/storage/v1/b/knative-prow/o/pr-logs%2Fpull%2Fknative-sandbox_eventing-kafka-broker%2F249%2Fpull-knative-sandbox-eventing-kafka-broker-integration-tests%2F1314175765880770561%2Fartifacts%2Fe6613896-095e-11eb-bc1f-be99e3e75c0a%2Fknative-eventing-logs%2Fknative-eventing%2Fsacura-wnqww-sacura.log?jk=AFshE3VxZDGjIc2Ay40UmLxpUCYqfUyp5wRP9RWSh99dNEgpTNGcpOctYM_5-15rPaEKHa6oyWXdr_rCri6yTdOcX0LhM8VCMphGWG2QK7b5TXwwWiS_GMJ6HLVVSQK2ZiT7MbupZKDzH6LgRtKLDUNiPO-u8LCxrawW5RC64gKBhbzeO69xZXaJkhhZyfcZeKAZpw641Dv6W0cS_yO4mvtIFSqz7oz4ST8kyBYbvqY-_U6BBMrMtTUPxaGkEmyg8QcWE2J4aqv-dQl0u7Z74Si6yHz0WtGewOWCdqjIIK9hvlav4eyviSYV11QaOPCXKT6l7U-q9ROIJib1V3iU9fLQrdFPcqIe2pfw79iNt6CAfwkC1CWbHBwUKo73o2tXYMbgzAx07NheBYxeHpt9XEENQMTauwceZs0su2kAQb5grQ0s5hK_yvzh_3yi1LrPSq3OspwedhL8WO86stY7_Oq2krF5INGm8eqpfcZzMj6HLYvzvnkyXxiNp2lKAdT90BJIRvfrpoR9XviywEqv0wbyMfNlj3tQqjcjmybG6ccWCszp2LaSIjcLZw_OVHKARZdrYYvfy-B0AGYyimDBYCs1bWnRwbImODjJt0bTMU1h_Vv0ByMY-slSrWeKX4b0BSCfP1dcxTp58kYQa7McfubeK0wC-SGGpT2PJKBJOElc7GU7eLZ2KcdXKb9b8wIM-GrUTCc5h5NrH-ubQAStYxS9kuNvzOP_BtAyOpEjZeDngAnpd-ssU92vsO15BRFVLefFk7k8aXX6Mc9ahVYiKFBQsC_V4l8pknuheGofDz88UeYql3zdg1aZqPmJgDhZZsFoqabqXVvIT5kLOHuHKYcOHu6kGaYl-lDDA7Y8ja7RDtqFaZ4CNvEQ6jfYdF4Kl9SyxBE2eu1dAZ0074HPlBXx3LqV9j0hTkRWi4vCsMumJrVgKJjBKGLZD9uUth-rMUPZwMdd4AsgfLJQrHeBex9bdvmS3YYi4gdL2uUBXqsW2M_DbPPQFcHVUMImhyNIFQwHEsfUhA-9rDb4tw73uTjsqnUSecDzQZ3JF2HIsTz3CxRbbSmVNSfKu6OYY3lHvxY5OWdi6YaNWhYW89TkAqkU9QSL5kJmKPOIlysXJgTDyM6wZSy0SugS3DQfmJ1BIULx6zMGsf4dDGWC5bADyyu2OzpD8qMHeVJiDYjooUZD0Uh4Qk40vncMtMZvfw9DAg8&isca=1), retesting
/test pull-knative-sandbox-eventing-kafka-broker-integration-tests

**Comment by pierDipi** [comment]
4/4
worked, retesting
/test pull-knative-sandbox-eventing-kafka-broker-integration-tests

**Comment by pierDipi** [comment]
5/5
worked, retesting
/test pull-knative-sandbox-eventing-kafka-broker-integration-tests

**Comment by pierDipi** [comment]
Worker, increasing frequency and duration.

**Comment by pierDipi** [comment]
Worked, increasing frequency to 50 and duration to 10 minutes.

**Comment by pierDipi** [comment]
Worked, retesting
/test pull-knative-sandbox-eventing-kafka-broker-integration-tests

**Comment by pierDipi** [comment]
```
Error from server (InternalError): error when creating "STDIN": Internal error occurred: failed calling webhook "validation.webhook.eventing.knative.dev": Post https://eventing-webhook.knative-eventing.svc:443/resource-validation?timeout=2s: EOF
```
https://github.com/knative/pkg/issues/1509

/retest

**Comment by pierDipi** [comment]
Worked server times, retesting one more time
/test pull-knative-sandbox-eventing-kafka-broker-integration-tests

**Comment by slinkydeveloper** [comment]
@pierDipi can you be more specific on what this tool does and why we need it?

**Comment by pierDipi** [comment]
The reason for adding this is to avoid introducing any regression regarding losing events.
The tool starts a sender and a receiver and expects that all events it sends should be received.
If after a configurable timeout it doesn't receive all events, it will log the id of all events it hasn't received (see [logs](https://00f74ba44b277e5af2f674ba283583fe622b3bb097-apidata.googleusercontent.com/download/storage/v1/b/knative-prow/o/pr-logs%2Fpull%2Fknative-sandbox_eventing-kafka-broker%2F249%2Fpull-knative-sandbox-eventing-kafka-broker-integration-tests%2F1312439860694355968%2Fartifacts%2F49e95551-059b-11eb-9e18-4ac9bffcbb9b%2Fknative-eventing-logs%2Fknative-eventing%2Fsacura-p5kcj-sacura.log?jk=AFshE3ULPZr3U8EXFE_xx_mcI3SOthS2WVvtXzEWV3kLLRuS9sKN3F5zuKDYEsiwbyg42KDgF5Brl5dMYrPoBXcafqrrwXYY4grEnIHRoy1hRu_EomRWaw_clKRcRwX5cuUC1J5P7AXqPiGW2hDGwHImk2d_37z_KH1K03ti3HZDkw9iHXH8-art68-g7faaCiV3zwfrYgkcMzrJ9PowoVMFeKovTOLoHsF-RHSvM1xc07A8PZMoKvhQiUyZ-PZH-B7OZ1KixVp3S0vc351J8rJVRuKGoYyHaLQc6B9gafX0Afi8at_pXu5zeMyZ12kf1j6c5jjPuHpBi0HrO3OS33ptFZwHbxK3ZmQpQwqgCnOwpb9AUg6IL3Aux9YWKSSy8JqDad1RsS_ZI_PaP_5TDdB3RNrb-PIfVWk9ndqUs3ruhN_XVq0ajSYwL16XsJqOE2VONkAkYlmAKF5a_q_XZ3yRbhudpPXoEBiTo-QzKVqbyKifBQB19ibdPfwYwpMe-ZQ8TL9QsKRSXtesvYra3i0OzUEsBaFjgl7ttvRV4qlyoHQ8dmh1JcXyMUEK8H29gjYfcJl1PxGgK9Sxal18jzrXZZpHvUGRgvsoPdmieM1Xm2xKCYjcWq6zWKma4KQt9zLr65ABXrtCuaraAVu5_JwpWfdLLV6nUvpw86B1VpUwI9V8ggfNY7Is2LCtRwSp8kmDNbUZHPP88KlHaa9TFW9eboH6rcgHWeYEsE0QwYKr6LtnqZ_DuT2L3b0O5qMBqv3bdZRXjVq0K2GNXnmPq19YWtghkaXxVVHT3baZwylHB8EagTNxPZUIwxbJHgQZ_ok2jApHt2-OaBbdNQcEr_H4tufy_3GizGXXmU_Gm4u6uBIaT6qq7uBUWAaFPeMP1eaWmLVaci4lg9fzHVJMcayROa9pGYUyRXcXz3NGh-dCat3bOHDnNUZlUdBTAsxmX9cbDoO3tqWVgEmsD46vhooOBkuoUJc1QguTBDyPndBxod5nLrQREd-QP0QSqfSosg89kknkiXn4-1ucHdMtHbwNMr2E7pEDFWQSk8eUk3XAAIpdIWGifkQb7FOhdqjeMaWprvpSyYsoDpZp2ZrUTrv-vdX3DrlX2hKZDq3DhOdSuzHzwl_bY9b4zBcS3735FLpD4pX26iFBnZZ3d-6WqHe4xybnIn2Dw2C3D-9Y2kvIJjGL-gSjARM8ohE22stGMfE&isca=1) for an example output).
Also, once we start implementing upgrade tests we can use the same tool to detect events loss during an upgrade or downgrade with a very simple flow:
- install version x
- create the topology (broker, sink, source, channel, flow, etc)
- start the tool in a k8s job
- upgrade to latest
- downgrade to x
- upgrade to latest
- verify 

Feel free to ask any other questions.

File: test/e2e-tests.sh [hunk 1]
```diff
@@ -22,10 +22,21 @@ apply_chaos || fail_test "Failed to apply chaos"
 
 header "Waiting Knative eventing to come up"
 
-wait_until_pods_running knative-eventing || fail_test "Knative Eventing did not come up"
+apply_sacura || fail_test "Failed to apply Sacura"
```
Comment (slinkydeveloper) [inline test/e2e-tests.sh:25]: Can we have a flag to enable/disable this thing when we run this script?

**Review by slinkydeveloper — COMMENTED** [review]
LGTM and definitely makes sense. My only question is: in the e2e tests script can we have a flag to opt-out from this load gen? It might be useful to avoid turning my Thinkpad in a burner when running the tests :laughing:

File: test/e2e-tests.sh (hunk 1 above)
Comment (pierDipi) [reply test/e2e-tests.sh]: Sure!

**Pull request merged**